src/utils/string.cljs

```diff
@@ -79,3 +79,22 @@
   (let [regex-pattern (str "^\\d*\\.?\\d{0," token-decimals "}$")
         regex         (re-pattern regex-pattern)]
     (boolean (re-matches regex amount-text))))
+
+(defn truncate-str-memo
+  "Given string and max threshold, trims the string to threshold length with `...`
+  appended to end or in the middle if length of the string exceeds max threshold,
+  returns the same string if threshold is not exceeded"
+  [s threshold & [middle?]]
+  (if (and s (< threshold (count s)))
+    (if middle?
+      (let [str-len   (count s)
+            max-len   (- threshold 3)
+            start-len (Math/ceil (/ max-len 2))
+            end-len   (Math/floor (/ max-len 2))
+            start     (subs s 0 start-len)
+            end       (subs s (- str-len end-len) str-len)]
+        (str start "..." end))
+      (str (subs s 0 (- threshold 3)) "..."))
+    s))
+
+(def truncate (memoize truncate-str-memo))
```

Comment:
In long-running systems it would be a problem to use `truncate` with the naive memoize function because it could indefinitely grow because the input strings can vary a lot (would be like a memory leak). I guess it's fine in our case?

There's no equivalent to https://github.com/clojure/core.memoize in CLJS that I know of, which is a shame, otherwise we would be able to use better memoization functions in many cases. It would be a fun project to for CLJS!

Reply:
Amount of tokens on coingecko is finite and around 10K, I don't think more than 10% have long names and there is a very low chance user will scroll them all. Even in this case, assuming that encoding is UTF-16 (2 bytes) and strings are around 40 characters, the memory consumption is:
```
40 chars × 2 bytes = 80 bytes per string
1000 strings × 80 bytes = 80,000 bytes = 78.13 KB
```

plus some overhead, overall 100KB, looks good enough.